AndelaOSP/art-backend
# [#156572526] Admin can create API Users

### What does this PR do ?
An admin can create api users
### Description of work done
- add django-oauth-toolkit to requirements
- add django-oauth-toolkit to base.py
- create custom way to add api user to models.py
- add permissions.py to limit to api users only
- write tests for feature
### How can I manually test this ?
- `python manage.py test` OR
- Login into the admin dashboard
- Add 1 or more`Security Users`
- Add an `API User`
- You can simply provide a name for the `API User` without filling any other details
- Click save
- On Postman allow the `API User` you created to access security users via giving it a token
   - As a `POST` request add this as the url `http://127.0.0.1:8000/api/v1/o/token/`
   - In the `Body` section use `form-data` and add the following keys
      - `grant_type` => `client_credentials`
      - `client_secret` => The client secret provided in the `API User` you created
      - `client_id` => The client id provided in the `API User` you created
      - Then click `Send`. You will get back an access_token, expiry time for the token, the token type and the scope of for the token provided
- Then as a `GET` request add this as the url `http://127.0.0.1:8000/api/v1/security-user-emails/`
- In the headers section add `Authorization` and in the value add the word `Bearer ` with a single space. Then add the `access_token` given to you in the `POST` response.
- Click `Send`. You should see a list of Security users.
### Relevant PT Story
[#156572526](https://www.pivotaltracker.com/story/show/156572526)

### Screenshots if appropriate
<img width="1170" alt="screen shot 2018-04-11 at 21 15 02" src="https://user-images.githubusercontent.com/6042152/38635829-41371a4e-3dcf-11e8-9588-ff30966d16da.png">

<img width="1127" alt="screen shot 2018-04-11 at 21 15 50" src="https://user-images.githubusercontent.com/6042152/38635843-4d5ef382-3dcf-11e8-9426-7aa0a4ef48b8.png">

<img width="1127" alt="screen shot 2018-04-11 at 21 16 22" src="https://user-images.githubusercontent.com/6042152/38635874-5fe45704-3dcf-11e8-99a9-63b90f91222c.png">

<img width="1119" alt="screen shot 2018-04-11 at 21 16 47" src="https://user-images.githubusercontent.com/6042152/38635881-64aeaf14-3dcf-11e8-9b72-800c345a8f8e.png">
